src/definition/populate.go

```diff
@@ -85,7 +85,14 @@ func IntegrationProtocol2PopulateFunc(
 					continue
 				}
 
-				ms := e.NewMetricSet(msType)
+				var ms metric.MetricSet
+				if len(e.Metrics) > 0 {
+					// This avoid generating a new ms for metrics collected from different sources like kubelet and ksm.
+					ms = e.Metrics[0]
+				} else {
+					ms = e.NewMetricSet(msType)
+				}
```

Comment:
How about:
```suggestion
				ms := e.NewMetricSet(msType)
				if len(e.Metrics) > 0 {
					// This avoid generating a new ms for metrics collected from different sources like kubelet and ksm.
					ms = e.Metrics[0]
				}
```

So we can avoid using `var`, explicit type declaration and `else` :smile:

Reply:
this was my first choise but then realized that when `e.NewMetricSet(msType)` is called , is being added to the entity already and that is exactly what i want it to avoid with this ~hack~ fix

Reply:
Make sense, then maybe:

```suggestion
                              if len(e.Metrics) == 0 {
				        e.Metrics = append(e.Metrics, e.NewMetricSet(msType))
					// This avoid generating a new ms for metrics collected from different sources like kubelet and ksm.
				}
				
				ms := e.Metrics[0]
```

It seems even better to the previous suggestion :smile:

Reply:
I like that! , I was not convinced about my if condition `len(e.Metrics) > 0` , but is liked the use of the else for readability, I almost never had the good chance to use it 😄

Reply:
had to modify your suggestion a bit since the append is already happening inside the New function

Reply:
Local DEVELOPER will do ANYTHING to avoid `else`